src/mail-app/mail/view/ConversationViewModel.ts

```diff
@@ -36,6 +37,7 @@ export class ConversationViewModel {
 	private loadingPromise: Promise<void> | null = null
 	/** Is not set until {@link loadConversation is finished. Until it is finished we display primary mail and subject. */
 	private conversation: ConversationItem[] | null = null
+	private allConversationMails: Mail[] | null = null
```

Comment:
I think I'm not fully understanding why we need this field. We can compute mails out of `ConversationItem` array, right?

I have a related concern also that this list is not getting updated with entity events. `MailItem` has a `MailViewerViewModel` which updates itself so we don't need to do any extra work.

I would suggest to not keep a separate array that can go out of sync but maybe I am missing something.